swimlane/core/resources/app_revision.py

```diff
@@ -0,0 +1,46 @@
+import pendulum
+
+from swimlane.core.resources.app import App
+from swimlane.core.resources.base import APIResource
+from swimlane.core.resources.usergroup import UserGroup
+
+
+class AppRevision(APIResource):
+    """Encapsulates a single revision returned from a History lookup."""
+
+    def __init__(self, swimlane, raw):
+        super(AppRevision, self).__init__(swimlane, raw)
+        self.version = App(swimlane, raw['version'])
+
+        self.modified_date = pendulum.parse(self._raw['modifiedDate'])
+        self.revision_number = self._raw['revisionNumber']
+        self.status = self._raw['status']
+
+        # UserGroupSelection, can't set as User without additional lookup
+        self.user = UserGroup(self._swimlane, self._raw['userId'])
+
+    def __str__(self):
+        return '{} ({})'.format(self.version, self.revision_number)
+
+    @staticmethod
+    def __separator():
+        """
+        Separator for unique ids. Unlikely to be found in application ids. Although technically we do not currently
+        validate app ids in the backend for specific characters so this sequence could be found.
+        """
+        return ' --- '
+
+    @staticmethod
```

Comment:
Don't see any reason other than unit tests to have these be static or public

Reply:
I'm currently using these static methods in the AppRevisionAdapter. Maybe they should be moved there, but then the AppRevision will have to call a method on the Adapter to get its unique id for get_cache_index_keys. Not sure which is cleaner but both seem off to me, can you think of a better way?

Reply:
Looking back at the way that caching is looked up, it was definitely written to expect the one arg input, so makes sense how you were trying to patch in the revision number to get it to respect that. 

I'm thinking that'd be the right place to look to fix that in order for the `App`/`AppRevision` classes to be merged and still respect the different version numbers, would think that cache lookup call could be updated to respect combinations of keys, but this is starting to feel overcomplicated.

Reply:
I agree that adjusting the cache implementation would be the best way to clean this up. But might be a bit overkill for now. Maybe best to just get rid of the caching hack here and add it in later if it's needed.